kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java

```diff
@@ -151,6 +152,9 @@ public String getActiveProcessImage(String containerId, long procInstId, String
                 populateSubProcessLink(containerId, activeNode, subProcessLinks);
             });
 
+            fullLogs.stream().filter(node -> ((org.jbpm.kie.services.impl.model.NodeInstanceDesc) node).getType() != START.getValue() && ((org.jbpm.kie.services.impl.model.NodeInstanceDesc) node).getType() != END.getValue())
```

Comment:
Wouldn't it now make sense to merge the `populateSubProcessLink` invocations into this single foreach?
Meaning, remove the following line from L148
```
                populateSubProcessLink(containerId, completeNode, subProcessLinks);
```
And also remove whole foreach starting on L151:
```
            activeLogs.forEach(activeNode -> {
                populateSubProcessLink(containerId, activeNode, subProcessLinks);
            });
 ```
 and keep just
 ```
 fullLogs.stream().forEach(node -> populateSubProcessLink(containerId, node, subProcessLinks));
 ```
 I think it would be the same results, wdyt?

Reply:
yes you are right, My original idea that set different colours to abort or error nodes, for now, it's not necessary.

Reply:
It turned out I was wrong. We need to process the activeLogs last, to make sure navigation to latest sub-process works as expected by JBPM-9821.
Could you please adapt the PR again, so that this functionality is maintained?

Reply:
ok,  I forget this one, I will updated,